io.openems.edge.levl.controller/src/io/openems/edge/levl/controller/LevlControlRequest.java

```diff
@@ -0,0 +1,140 @@
+package io.openems.edge.levl.controller;
+
+import com.google.gson.JsonObject;
+import io.openems.common.exceptions.OpenemsError;
+import io.openems.common.exceptions.OpenemsError.OpenemsNamedException;
+import io.openems.common.jsonrpc.base.JsonrpcRequest;
+import io.openems.common.utils.JsonUtils;
+
+import java.time.Clock;
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+public class LevlControlRequest {
```

Comment:
What's the motivation to perform parts of the controlling via an external API and not within OpenEMS?

Reply:
Yes, it's about virtual power plants. I can imagine this being implemented within OpenEMS, with a virtual power plant coordinator running OpenEMS and multiple batteries connected to the power plant through OpenEMS. That’s the reason for my question.

This question isn’t intended to block or hinder the merge request; it’s simply out of curiosity.

Reply:
Hey guys, sorry for the late response. Levl enables battery owners to participate in the electricity market, making more efficient use of the batteries and reducing amortisation costs. At the same time, the electricity grid is relieved as electricity is bought when it is cheap (a lot of renewables energy in the grid) and sold when demand is high (coal and gas is needed).

Unfortunately, this is not quite so simple from a regulatory perspective, as battery storage systems behind-the-meter, especially with a primary use case such as optimising self-consumption, requires a distinction to be made between self-consumption and temporarily stored energy. This is the only way the energy supplier can correctly bill the quantities purchased. For this reason, we act as an aggregator with Levl.

In the end, this controller is comparable to various other controllers that make it possible to use an external service on the hardware and/or software side to optimise your own system.

To come back to your idea @parapluplu: If regulation were left out of the equation, that might be possible. But you would still need someone to run and manage this coordinator centrally. However, Levl's long-term goal is to enable all battery owners, regardless of the EMS used, to become part of a virtual power plant with their system. 
At the moment we're working with a lot of battery owners who are using OpenEMS as well as working directly with Fenecon. For this reason we're looking forward to becoming part of the OpenEMS community :-)